clients/include/testing_trsm_batched.hpp

```diff
@@ -0,0 +1,403 @@
+        for(int b = 0; b < batch_count; b++)
+        {
+            max_res_1 = max_res_2 = 0;
+            for(int i = 0; i < N; i++)
+            {
+                T res_1 = 0.0;
+                T res_2 = 0.0;
+                for(int j = 0; j < M; j++)
+                {
+                    int idx = j + i * ldb;
+                    if(hB[b][j + i * ldb] != 0)
+                    {
+                        res_1 += std::abs((hXorB_1[b][idx] - hB[b][idx]) / hB[b][idx]);
+                        res_2 += std::abs((hXorB_2[b][idx] - hB[b][idx]) / hB[b][idx]);
+                    }
+                    else
+                    {
+                        res_1 += std::abs(hXorB_1[b][idx]);
+                        res_2 += std::abs(hXorB_2[b][idx]);
+                    }
+                }
+                max_res_1 = max_res_1 > res_1 ? max_res_1 : res_1;
+                max_res_2 = max_res_2 > res_2 ? max_res_2 : res_2;
+            }
+            trsm_err_res_check<T>(max_res_1, M, residual_eps_multiplier, eps);
+            trsm_err_res_check<T>(max_res_2, M, residual_eps_multiplier, eps);
+        }
+    }
+
+    if(arg.timing)
+    {
+        double gpu_time_used, cpu_time_used;
+        double rocblas_gflops, cblas_gflops;
+
+        // GPU rocBLAS
+        for(int b = 0; b < batch_count; b++)
+        {
+            CHECK_HIP_ERROR(
+                hipMemcpy(XorBv[b], hXorB_1[b], sizeof(T) * size_B, hipMemcpyHostToDevice));
+        }
+        CHECK_HIP_ERROR(hipMemcpy(dXorB, hXorB_1, sizeof(T*) * batch_count, hipMemcpyHostToDevice));
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+
+        gpu_time_used = get_time_us(); // in microseconds
+
+        CHECK_ROCBLAS_ERROR(rocblas_trsm_batched<T>(
+            handle, side, uplo, transA, diag, M, N, &alpha_h, dA, lda, dXorB, ldb, batch_count));
+
+        gpu_time_used  = get_time_us() - gpu_time_used;
+        rocblas_gflops = batch_count * trsm_gflop_count<T>(M, N, K) / gpu_time_used * 1e6;
+
+        // CPU cblas
+        cpu_time_used = get_time_us();
+
+        for(int b = 0; b < batch_count; b++)
+            cblas_trsm<T>(side, uplo, transA, diag, M, N, alpha_h, hA[b], lda, cpuXorB[b], ldb);
+
+        cpu_time_used = get_time_us() - cpu_time_used;
+        cblas_gflops  = batch_count * trsm_gflop_count<T>(M, N, K) / cpu_time_used * 1e6;
+
+        // only norm_check return an norm error, unit check won't return anything
+        std::cout << "M,N,lda,ldb,side,uplo,transA,diag,rocblas-Gflops,us";
+
+        if(arg.norm_check)
+            std::cout << ",CPU-Gflops,us,norm_error_host_ptr,norm_error_dev_ptr";
+
+        std::cout << std::endl;
+
+        std::cout << M << ',' << N << ',' << lda << ',' << ldb << ',' << char_side << ','
+                  << char_uplo << ',' << char_transA << ',' << char_diag << ',' << rocblas_gflops
+                  << "," << gpu_time_used;
+
```

Comment:
Probably want to add batch_count to the timing print of each testing_batched file.

Reply:
Done, 9d378b0.